modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java

```diff
@@ -386,12 +458,52 @@ private ComponentModel innerFindComponentDefinitionModel(Element element, List<C
     }
 
     /**
+     * TODO MULE-9688: When the model it's made immutable we will also provide the parent component for navigation and this will not be needed anymore.
+     * @return the root component model
+     */
+    public ComponentModel getRootComponentModel()
+    {
+        return muleComponentModels.get(0);
+    }
+
+    /**
      * Functional interface to process a component.
      */
     @FunctionalInterface
-    interface ComponentConsumer
+    public interface ComponentConsumer
     {
 
         void consume(ComponentModel componentModel) throws MuleRuntimeException;
```

Comment:
makes sense to declare a runtime exception? Also, why don't just use a java `Consumer<ComponentModel>`?